[Redesign add agent] Integration commands generator with UI

Machi3mfl opened the pull request:
### Description

This PR adds the integration of the command generator and the add new agent UI (form).

Closes #5547


### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

Machi3mfl commented:
# Test cases

## 1 - Check that commands appears when `OS` is selected and `Server address` is completed and valid.

- Go to add new Agent section
- Click on every OS checkbox
- Complete the server address input
- Check that the install and start command sections are enabled
- Complete the optional parameters and check if the values are shown in the command
- Repeat this case for every OS and check that the commands (Install and start) change.
  - The commands are defined in https://github.com/wazuh/wazuh-kibana-app/blob/feat/5547-integration-commands-ui/public/controllers/register-agent/core/config/os-commands-definitions.ts





## 2 - Check that command steps should be disabled when `OS` is not selected or `Server address` is incomplete/invalid.

- Go to add new Agent section
- Click on every step or leave it empty
- Complete the server address input with an invalid value
- The command steps should be disabled with an empty code block

### A - Empty form fields
![image](https://github.com/wazuh/wazuh-kibana-app/assets/6089438/dd42cced-9cfa-417f-87c2-cb5031c87c75)

### B - Empty server address input 
![image](https://github.com/wazuh/wazuh-kibana-app/assets/6089438/cffccaa6-36f4-4441-97ed-1aab5438dd9a)

### C - Invalid server address input 
![image](https://github.com/wazuh/wazuh-kibana-app/assets/6089438/aa44ffc3-eb70-4cde-b39e-cc4416913c74)

### D - OS not selected

![image](https://github.com/wazuh/wazuh-kibana-app/assets/6089438/35f65c10-7b55-46fd-8634-bde64081a810)


**Important: 
No validate the commands, only check the form behavior and if the filled inputs are shown in the command code input**

yenienserrano reviewed (commented):
🟡 
I have more than 1 group selected, but only the first one selected is displayed. 
![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/719e4305-6b48-4ad7-98d0-0d20c005b0e5)

<details><summary>Details</summary>

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/e568e75e-c1c5-45b4-a754-04c60d484e74)

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/eecda68a-446e-4710-9f40-ffdc5ac7b125)

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/72f03b92-f8a2-497d-942c-5024c0948d37)

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/4b35f78e-b366-40f7-b88f-6b278e71a363)

</details>

yenienserrano commented on public/controllers/register-agent/containers/steps/steps.tsx:
```diff
@@ -1,143 +1,104 @@
+      optionalParams: {
+        agentGroups: '',
+        agentName: '',
+        serverAddress: '',
```
If this is necessary for the installation code to appear, I do not see it as an optional parameter.

Machi3mfl replied:
Solved in https://github.com/wazuh/wazuh-kibana-app/pull/5593/commits/57f27664b2a37972f16f51fa6db27cf1d7dfda79

yenienserrano commented on public/controllers/register-agent/containers/steps/steps.tsx:
```diff
@@ -1,143 +1,104 @@
+        architecture: '',
+      },
+      optionalParams: {
+        agentGroups: '',
```
This should probably be an array

yenienserrano commented on public/controllers/register-agent/core/config/os-commands-definitions.ts:
```diff
@@ -0,0 +1,209 @@
+    property: 'WAZUH_AGENT_GROUP',
+    getParamCommand: props => {
+      const { property, value } = props;
+      return value !== '' ? `${property}=${value}` : '';
```
This should probably be an array

chantal-kelm commented:
Double space in front of the WAZUH_AGENT_GROUP variable when only the mandatory steps have been completed.

<details><summary>Details</summary>
<img width="889" alt="Captura de pantalla 2023-06-26 a la(s) 12 34 50" src="https://github.com/wazuh/wazuh-kibana-app/assets/99441266/347b8087-785a-4ff2-a738-0c2ba544d146">

yenienserrano commented:
The group variable is displayed even if no group is configured.

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/1a13ce32-c4c3-409e-916c-9ec7317b6b67)

yenienserrano reviewed (approved):
LGTM 

As an enhancement to the overview, it would be nice to be able to click on the name of the operating system and select the system.

![image](https://github.com/wazuh/wazuh-kibana-app/assets/63758389/52fd1c13-d0d6-4cf6-8d91-472d26f1c020)

Machi3mfl commented:
## @chantal-kelm Requested changes
### 1 - Commands formats
- Double space in front of the WAZUH_AGENT_GROUP variable when only the mandatory steps have been completed.
- When it is an IPV6 or IPV4 it paints the first numerous of the IP of the color of the variable named WAZUH_MANAGER.

  ### The correct commands format will be added with the issue: https://github.com/wazuh/wazuh-kibana-app/issues/5594

### 2 - Wrong WAZUH_AGENT_GROUP display

- The variable WAZUH_AGENT_GROUP is displayed even if it does not contain any value.

Fixed in https://github.com/wazuh/wazuh-kibana-app/pull/5593/commits/1c1b767514f69e5482570a62dccebeacfbd51f1f

### 3 - Wrong form steps statuses

- When copying the step 4 command, step 4 is not marked as completed, it still appears as current.
- When copying the step 5 command, step 5 is not marked as completed, it still appears as current.

Fixed in https://github.com/wazuh/wazuh-kibana-app/pull/5593/commits/1c1b767514f69e5482570a62dccebeacfbd51f1f

### If any form  step is incomplete or invalid the steps of the command will be shown as disabled

### 4 - Code Editor style 
- The styles of the commands in steps 4 and 5 are not correct.

## I think is better not to change the code editor style to prevent future errors and get the code editor colors when recognizing some syntaxes

![image](https://github.com/wazuh/wazuh-kibana-app/assets/6089438/9a04cb97-d64d-4819-ab4b-eda6a1acad09)

jbiset commented on public/controllers/register-agent/core/register-commands/exceptions/index.ts:
```diff
@@ -1,7 +1,7 @@
+  constructor(osName: string, architecture: string) {
     super(
-      `No OS option found for "${osName}" "${architecture}" "${extension}".Please check the OS definitions."`,
+      `No OS option found for "${osName}" "${architecture}".Please check the OS definitions."`,
```
Missing space before "Please check..."

jbiset reviewed (approved):
## Evidence
**1 - Check that commands appears when OS is selected and Server address is completed and valid.**
https://github.com/wazuh/wazuh-kibana-app/assets/43619595/8b09dafc-72f7-40cc-81f3-04747db5a38e

**2 - Check that command steps should be disabled when OS is not selected or Server address is incomplete/invalid.**
**A - Empty form fields**
![Check2-A](https://github.com/wazuh/wazuh-kibana-app/assets/43619595/5f9f5489-0d32-4506-8367-30f96448cb5b)

**B - Empty server address input**
![Check2-B](https://github.com/wazuh/wazuh-kibana-app/assets/43619595/6b7a1fb9-f741-418b-953a-d32f7cc5d959)

**C - Invalid server address input**
![Check2-C](https://github.com/wazuh/wazuh-kibana-app/assets/43619595/062c4162-bfbf-4360-a0d7-6f069c12944e)

**D - OS not selected**
![Check2-D](https://github.com/wazuh/wazuh-kibana-app/assets/43619595/128d9770-2f12-40ae-abbe-8c889f1ad5b0)

Desvelao commented on public/controllers/register-agent/containers/steps/steps.tsx:
```diff
@@ -1,143 +1,104 @@
+  connection,
+  wazuhPassword,
+}: IStepsProps) => {
+  const [registerAgentFormValues, setRegisterAgentFormValues] =
+    useState<IParseRegisterFormValues>({
+      operatingSystem: {
+        name: '',
+        architecture: '',
+      },
+      optionalParams: {
+        agentGroups: '',
+        agentName: '',
+        serverAddress: '',
+        wazuhPassword,
+      },
+    });
```
**thought**: The form state is controlled, why do we need to handle of this way? Can not we use the form state instead?

Machi3mfl replied:
This is because the Command Generator i need to pass only the values to the Command Generator. And must be passed separated by OS and optionals params.

Desvelao commented on public/controllers/register-agent/containers/steps/steps.tsx:
```diff
@@ -1,143 +1,104 @@
+  useEffect(() => {
+    if (
+      registerAgentFormValues.operatingSystem.name !== '' &&
+      registerAgentFormValues.operatingSystem.architecture !== ''
+    ) {
+      selectOS(registerAgentFormValues.operatingSystem as tOperatingSystem);
+    }
 
-  const onButtonAgentName = () =>
-    setIsPopoverAgentName(isPopoverAgentName => !isPopoverAgentName);
-  const closeAgentName = () => setIsPopoverAgentName(false);
+    setOptionalParams(registerAgentFormValues.optionalParams);
+  }, [registerAgentFormValues]);
```
**thought**: if we use the form state to generate the related install and start command, instead of saving in a new state, we will avoid a rerender.

chantal-kelm commented:
As soon as you complete the ip all steps are marked as completed:

:red_circle: Item 3 should be checked as completed once it is completed or once the user copies the command from step 4.

<details><summary>Details</summary>

https://github.com/wazuh/wazuh-kibana-app/assets/99441266/6782ebf4-a0c9-40d7-95d8-693cbdec5d67


</details>

:red_circle: Step 4 should be checked as completed when the user copies the command.

<details><summary>Details</summary>

https://github.com/wazuh/wazuh-kibana-app/assets/99441266/2c342ba8-a8c1-4938-bfcc-33305c090dcc


</details>

:red_circle: Step 5 should be checked as completed when the user copies the command.

<details><summary>Details</summary>

https://github.com/wazuh/wazuh-kibana-app/assets/99441266/5ba977f1-d23f-4dd6-ae29-e4441a014aee


</details>

chantal-kelm commented:
:red_circle: Item 3 should be checked as completed once it is completed or once the user copies the command from step 4.

<details><summary>Details</summary>


https://github.com/wazuh/wazuh-kibana-app/assets/99441266/1cc682a5-554c-4ed2-ae26-2db0e048eec0




</details>

Machi3mfl commented:
### Fixed form steps status

## Current solution.

- When any previous step is `INVALID` or `INCOMPLETE` the following step status is `DISABLED`
- When we are editing any input if the previous steps are `COMPLETE` we current step is `CURRENT`
- The step is `COMPLETE` when is not empty and has a valid value (except for COMMANDS steps)
- The command s

https://github.com/wazuh/wazuh-kibana-app/assets/6089438/6fcbf90c-c1ea-4b49-adc1-adc7cd2090d9

teps will be `COMPLETE` only after the user click on the `copy button`.
